### ISSUE TITLE
Revamp website with new UI

### ISSUE THREAD

**Issue by damianopetrungaro** [conventional-commits/conventionalcommits.org#81]
What do you think about revamping the UI of the website with a new theme (and\or new tool for generating the static website)?

We can make it as cool as [keep a changelog](http://keepachangelog.com/en/1.0.0/) so that more people will read it because it looks good

**Comment by damianopetrungaro** [comment]
Looking for sth like this: https://themes.gohugo.io/theme/hugo-icon/

**Comment by hbetts** [comment]
I have no particular preference for a theme, though I am fine with y'all going with a new theme.

My only request is that the following be taken into consideration:
- The theme should be [web accessible](https://www.w3.org/WAI/fundamentals/accessibility-intro/).
- The tool needed to generate a website from a them should be low friction.
- The theme should emphasize the content of the standard (and not detract from it with flashy graphics or other, unrelated content)

**Comment by damianopetrungaro** [comment]
Yeah sure, I was thinking to use `hugo` because it has more themes (and it's also faster to compile and reload changes, but this is not a big deal for this project).

Feel free to suggest any theme for any tool you like and of course match what @hbetts  recommend.

**Comment by damianopetrungaro** [comment]
![example](https://user-images.githubusercontent.com/8950503/44368808-65a2f300-a4d4-11e8-82ed-cc6ded140bf8.gif)

Generated using a file [like this](https://gist.github.com/damianopetrungaro/902091136592514750be4f6003f67bbf)

Powered by Hugo and [hugo-theme-sarah](https://github.com/avelino/hugo-theme-sarah)
Of course, I still need to change some colors, links add images etc. etc. but this one looks better IMHO.

Let me know about it ;)

**Comment by hbetts** [comment]
It looks fine to me.

**Comment by damianopetrungaro** [comment]
Please, @conventional-commits/committee gives me a feedback so I will put some effort if all of you are fine with it ;)

**Comment by damianopetrungaro** [comment]
I got some free time at home and i designed those two things.

If we are going to implement a custom design we need someonw who will write HTML and CSS ( i never wrote it so i need external help)

Feedbacks appriciated:

<img width="882" alt="screen shot 2018-08-21 at 11 31 17 pm" src="https://user-images.githubusercontent.com/8950503/44430408-e0364600-a59a-11e8-8198-af994b9a2f0b.png">

<img width="881" alt="screen shot 2018-08-21 at 11 32 00 pm" src="https://user-images.githubusercontent.com/8950503/44430409-e0364600-a59a-11e8-9a26-222acde0c92f.png">

**Comment by hbetts** [comment]
> If we are going to implement a custom design we need someonw who will write HTML and CSS

For that reason I would prefer if we limited how _much_ customization we do. Unless we can find two or more people that are able and willing to dedicate the time to being our HTML/CSS contributors.

As for the designs in your latest commit, I think they look great. My only suggestion would be to increase the font size so that the text is easier to read.

**Comment by damianopetrungaro** [comment]
Already done ;)
> As for the designs in your latest commit, I think they look great. My only suggestion would be to increase the font size so that the text is easier to read.

This will be one very simple page so the effort won't be huge.
If someone is free to spend some time coding it would be great!

> For that reason I would prefer if we limited how much customization we do. Unless we can find two or more people that are able and willing to dedicate the time to being our HTML/CSS contributors.

**Comment by damianopetrungaro** [comment]
Found someone who has some free time next week!

@lorenzodianni is the best js developer I have ever known and he uses the conventional commits for months!

He'd be happy to take care of the HTML/CSS page.
And it would be great also having your feedback about the design and the project itself!

**Comment by zeke** [comment]
Wow really cool to see this design work, @damianopetrungaro 🎨 

> new tool for generating the static website)?

I haven't used [hugo](https://gohugo.io/), though I know it's popular and active. I wouldn't be opposed to this if you want to take the initiative, @damianopetrungaro 👍 

That said, when pondering how I personally would revamp the existing site, I imagined using Node.js (rather than Ruby or Go) and a set of of "[small sharp modules](https://gist.github.com/adamwiggins/5687294#small-sharp-tools)" like:

- https://ghub.io/hubdown to convert markdown into GitHub-style HTML
- https://ghub.io/github-markdown-css to style that HTML
- https://ghub.io/nanohtml to wrap pages in a layout.

The above tools could be used to generate a static site and/or a dynamic server-rendered site.

**Comment by damianopetrungaro** [comment]
I chose Hugo for the great doc and the super active support.

> That said, when pondering how I personally would revamp the existing site, I imagined using Node.js (rather than Ruby or Go) and a set of of "small sharp modules" like:

https://ghub.io/hubdown to convert markdown into GitHub-style HTML
https://ghub.io/github-markdown-css to style that HTML
https://ghub.io/nanohtml to wrap pages in a layout.
The above tools could be used to generate a static site and/or a dynamic server-rendered site.

**Comment by bcoe** [comment]
@zeke @damianopetrungaro I like the direction we're going in, but I don't love that I need to click a button to start reading the spec, it seems like the page you land on is a bit barren. 

Maybe the tldr; description of the specification could be above the fold, and then we dig into it deeper as you scroll down the page.

With this new design, the actual specification itself will still be in markdown?

**Comment by damianopetrungaro** [comment]
@bcoe the idea is to scroll and read the spec, what I posted is just the first section the users will see.

>Maybe the tldr; description of the specification could be above the fold, and then we dig into it deeper as you scroll down the page.

It depends if we wanna change it, I already linked an example of how the doc could be changed so.
But if we don't want to move from markdown,  we won't.
The new design will just be a UI improvement.
> With this new design, the actual specification itself will still be in markdown?

**Comment by damianopetrungaro** [comment]
Yup, this is what exactly I am working on 😄

**Comment by damianopetrungaro** [comment]
@zeke I've done a small test on how it may look like.

https://github.com/damianopetrungaro/_hugotest
https://github.com/damianopetrungaro/_hugotest/blob/master/content/beta0.0.3/index.md

Quick explanation:

`{{< welcome >}}`
will load the first section (the one with the nice UI linked ⬆️ )

`{{% white-partial %}}...{{% white-partial %}}`
will inject the markdown defined in the two shortcodes in some HTML defined in the theme ([see here](https://github.com/damianopetrungaro/_hugotest/tree/master/themes/conventional-commits/layouts/shortcodes))

**Comment by hbetts** [comment]
Purely my _two cents_ based on my aesthetics preferences:
1. The second color palette looks to washed out.
2. The commit message structure in the _Summary_ section, and the examples in the _Examples_ section need the be more distinguished (They kind of blend in with the surrounding text).
3. The second _Specification_ section needs to be called _FAQ_ or _Frequency Asked Questiones_.
4. Why are there two _Read the Spec_ buttons at the top?

**Comment by damianopetrungaro** [comment]
1. If you don't like it we won't use that one  :)
2. I was thinking to use a code block there
3. Yep, my bad 
4. One is the default one, the other one the mouse over :)

**Comment by damianopetrungaro** [comment]
If no one has nothing to add on this I'll ask to @lorenzodianni to start working on CSS and HTML

**Comment by bcoe** [comment]
@damianopetrungaro I'm really happy with how this is looking.

**Comment by lorenzodianni** [comment]
Hi to everyone!
@damianopetrungaro send me all info about the new style, i'll work on it asap! 🕺

**Comment by damianopetrungaro** [comment]
Sure 💪

**Comment by damianopetrungaro** [comment]
Thanks to the great effort and the great work that @lorenzodianni put into it, we are proud to announce the first demo of the UI revamp!

https://conventional-commits-test.netlify.com/en/v1.0.0-beta.2/

@hbetts  @zeke @bcoe 
There are still a couple of things to optimize (like the redirects and the anchor link to the specs), but this will be super close to the final version!
Any feedback is welcome!

Code here https://github.com/damianopetrungaro/_hugotest

**Comment by damianopetrungaro** [comment]
FYI:
We'll have for free (0 configurations and 0$) CD on Pull Request thanks to [netlify](http://netlify.com/)

**Comment by hbetts** [comment]
Three things I came across:

First, the drop-down for _Versions_  is really difficult for me to use. Hovering over a version for more than a second causes the menu to vanish. (Using latest Firefox)

Second, the title looks cutoff for the specification:

![screenshot_20180904_091235](https://user-images.githubusercontent.com/6701030/45036761-faecce00-b022-11e8-903f-d710b611cccd.png)

^ That's what I see at the very bottom of my screen when I first load the page.

Instead of just saying _GitHub_, I feel like changing it to _Contribute_ may help encourage our community to take a more active role in the development of this specification.

**Comment by hbetts** [comment]
It should be said, though, that I really do like the website. Contrast feels pretty good. It's easy to scroll through and find content. Branding is really starting to take shape.

:+1:

**Comment by damianopetrungaro** [comment]
Sure!
> Instead of just saying GitHub, I feel like changing it to Contribute may help encourage our community to take a more active role in the development of this specification.


@lorenzodianni let's fix that https://github.com/conventional-commits/conventionalcommits.org/issues/81#issuecomment-418383793

**Comment by damianopetrungaro** [comment]
@hbetts  i am not a frontend dev, but doesn't the content shown at the very bottom of the screen when the page is loaded depends on the screen size?

**Comment by damianopetrungaro** [comment]
**I am also super proud to announce that [netlify](https://netlify.com) gave us a special pro plan for free!**

We'll be able to add more than collaborators to our team, and enjoy a free CD for each new PR and branch!

I'd like to add them in the README when the new UI we'll go live!

@conventional-commits/committee  Please send me your email in the private team and I'll add you on the netlify platform!

**Comment by lorenzodianni** [comment]
@bcoe what do you think about this arrow?
<img width="138" alt="schermata 2018-09-05 alle 22 28 25" src="https://user-images.githubusercontent.com/7217805/45119315-66d25200-b15b-11e8-8639-929076bdf2d4.png">
___
@hbetts about specification title:
the gradient section has always 100% viewport height and the article section has a negative translationY (so the article moves up).
Do you think it's better to move it higher or to move it under the gradient section? (we will see the article only after scroll)

**Comment by damianopetrungaro** [comment]
@lorenzodianni maybe see the article only after scrolling?

**Comment by hbetts** [comment]
> negative translationY (so the article moves up).

Thank you for the explanation! 👍 

> Do you think it's better to move it higher or to move it under the gradient section?

That's a really tough one.

I moved the content up using -95px and it feels like it's crowding the content on the splash page. Also, it seems a little redundant (The title of the splash page, and the title of the article both say _Conventional Commits_).

Moving the content down looks nicer, but it's not apparent that you can scroll down to read additional content.

Any thoughts on that later point?

**Comment by damianopetrungaro** [comment]
All the changes have been applied.
See that I have no more feedbacks I'll open a PR today, and pair with @bcoe to transfer the domain to Netlify.

fyi: `https://conventional-commits-test.netlify.com/en/v1.0.0-beta.2/`
